### ISSUE TITLE
React Router proxy querystring issue

### ISSUE THREAD

**Issue by maxsalven** [react-cosmos/react-cosmos#476]
When clicking on a `Link` in a component that uses react-router, if the link has a query string, then the component will render appropriately but the `fixture.url` value will not include the query string in the editor.

As a basic fix, we should render the url correctly in the editor. I haven't looked into it, but this may be as simple as changing:
https://github.com/react-cosmos/react-cosmos/blob/master/packages/react-cosmos-router-proxy/src/LocationInterceptor.js#L5
`const getPathFromProps = props => props.location.pathname;`
to
``const getPathFromProps = ({location: { pathname, search, hash }}) => `${pathname}${search}${hash}`;``

As a more advanced fix, we should support the `location` object from react-router, which includes a state field that is not exposed by the url but can impact how your component renders:
https://github.com/ReactTraining/react-router/blob/master/packages/react-router/docs/api/location.md

I'm not sure what the best way to proceed here is.

One option is to allow specifying only either `fixture.url` or `fixture.location`. This is probably the simplest solution, but isn't that nice.

I'd prefer to see both when I use the react-router proxy and for the two to stay in sync when I navigate the site. But when I set up my fixture, I'd prefer to just specify something like:

```
{
  url: "/my/url?foo=true#somehash",
  location: {
    state: {
      from: "dashboard"
    }
  }
}
```

and then when I open the fixture in cosmos, I get a fully populated location:
```
{
  url: "/my/url?foo=true#somehash",
  location: {
    pathname: "/my/url",
    search: "?foo=true",
    hash: "#somehash",
    state: {
      from: "dashboard"
    }
  }
}
```

It's a little bit of magic, but I don't think it will confuse anyone.

The main question here is what if you edit the fixture in the react-cosmos fixture editor, and you put in contradictory information (i.e. location.pathname doesn't match what's in location.url). Ideally the change you make would get reflected in the appropriate other field.

I'll try to find time to play around with the router and see what works.

**Comment by skidding** [comment]
Thanks for opening this. My first thought is that synchronizing `fixture.url` and `fixture.location` is a world of pain. The only clean option for keeping `pathname`, `search` and `hash` together while preserving the location state is something like this:

```js
{
  url: "/my/url?foo=true#somehash",
  locationState: {
    from: "dashboard"
  }
}
```

The router proxy can map `url` and `locationState` deterministically to the Router's `location` structure and vice versa.

**Comment by skidding** [comment]
Closed by #477
